src/contactmanagementsoftware/MUI.java

```diff
@@ -194,18 +194,32 @@ public void setDescription() {
     public final void setUpTableData() {
         DefaultTableModel tableModel = (DefaultTableModel) jXTable1.getModel();
         tableModel.setRowCount(0);
-        ArrayList<Acquaintances> list;
-        try {
-            list = acquaintances.get(Categories.getSelectedIndex());
-        } catch (Exception e) {
-            return;
-        }
-        for (int i = 0; i < list.size(); i++) {
+//        ArrayList<Acquaintances> list;
+//        try {
+//            list = acquaintances.get(Categories.getSelectedIndex());
+//        } catch (Exception e) {
+//            return;
+//        }
+//        for (int i = 0; i < list.size(); i++) {
+//            String[] data = new String[4];
+//            data[0] = Integer.toString(i + 1);
+//            data[1] = list.get(i).getName();
+//            data[2] = list.get(i).getMobileNo();
+//            data[3] = list.get(i).getEmail();
+//            tableModel.addRow(data);
+//        }
+
+//        andre
+
+        Iterator iter = getIterator();
+        int index = 0;
+        while(iter.hasNext()) {
+            Acquaintances item = (Acquaintances) iter.next();
             String[] data = new String[4];
-            data[0] = Integer.toString(i + 1);
-            data[1] = list.get(i).getName();
-            data[2] = list.get(i).getMobileNo();
-            data[3] = list.get(i).getEmail();
+            data[0] = Integer.toString(index+1);
```

Comment:
not sure but do you need ++index?

data[0] = Integer.toString(++index);

Reply:
yeah ++index xD